### PR TITLE
Add emulated tests for the Coretime Interface calls to ensure sufficient weights

### PR DESCRIPTION
**Pull request by seadanda**
Add emulated test cases for the coretime chain.

This tests the calls sent across the `CoretimeInterface` and ensures the weights are sufficient.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5391749

File: cumulus/parachains/integration-tests/emulated/tests/coretime/coretime-rococo/src/tests/coretime_interface.rs [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+		// Hooks don't run in emulated tests - workaround.
+		<CoretimeRococo as CoretimeRococoPallet>::Broker::on_initialize(
+			<CoretimeRococo as Chain>::System::block_number(),
+		);
+
+		assert_expected_events!(
+			CoretimeRococo,
+			vec![
+				CoretimeEvent::MessageQueue(
+					pallet_message_queue::Event::Processed { success: true, .. }
```
Comment (muharem) [inline cumulus/parachains/integration-tests/emulated/tests/coretime/coretime-rococo/src/tests/coretime_interface.rs:214]: I wonder where is the sent event for this processed message?

File: cumulus/parachains/integration-tests/emulated/tests/coretime/coretime-rococo/src/tests/coretime_interface.rs [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+		assert_expected_events!(
+			Rococo,
+			vec![
+				RelayEvent::MessageQueue(
+					pallet_message_queue::Event::Processed { success: true, .. }
+				) => {},
+				RelayEvent::MessageQueue(
+					pallet_message_queue::Event::Processed { success: true, .. }
+				) => {},
```
Comment (muharem) [inline cumulus/parachains/integration-tests/emulated/tests/coretime/coretime-rococo/src/tests/coretime_interface.rs:195]: minor, but we probably can trigger Rococo between two CoretimeRococo above, to process these two messages one by one. I think that would be closer to what happens in prod.

**Review by muharem — APPROVED** [review]

File: cumulus/parachains/integration-tests/emulated/tests/coretime/coretime-rococo/src/tests/coretime_interface.rs (hunk 2 above)
Comment (seadanda) [reply cumulus/parachains/integration-tests/emulated/tests/coretime/coretime-rococo/src/tests/coretime_interface.rs]: Good point, done in c4ab8443ca0

File: cumulus/parachains/integration-tests/emulated/tests/coretime/coretime-rococo/src/tests/coretime_interface.rs (hunk 1 above)
Comment (seadanda) [reply cumulus/parachains/integration-tests/emulated/tests/coretime/coretime-rococo/src/tests/coretime_interface.rs:226]: DMP messages don't emit an event like UMP messages when sent. Emulated tests don't get the XcmPallet sent events for some reason.

**Review by kianenigma — APPROVED** [review]

**Pull request merged**